app/assets/javascripts/discourse/app/helpers/directory-item-user-field-value.js

```diff
@@ -1,11 +1,14 @@
 import { htmlSafe } from "@ember/template";
+import { generateUsersLink } from "discourse/helpers/generate-users-link";
+
+export default function directoryItemUserFieldValue({ item, column }) {
+  const userFields = item?.user?.user_fields;
+  const fieldData = userFields ? userFields[column.user_field_id] : null;
```

Comment:
May be able to condense the optional chaining further with bracket notation as well:

```suggestion
  const fieldData = item?.user?.userFields?.[column.user_field_id];
```